### PR TITLE
mapToOneOrNull replaced by mapToOneOrDefault

### PR DESCRIPTION
**Pull request by phajduk**
Solves #55

### CODE REVIEW

File: sqlbrite/src/main/java/com/squareup/sqlbrite/QueryToOneOperator.java [hunk 1]
```diff
@@ -9,11 +9,11 @@
 
 final class QueryToOneOperator<T> implements Observable.Operator<T, SqlBrite.Query> {
   private final Func1<Cursor, T> mapper;
-  private final boolean emitNull;
```
Comment (JakeWharton) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/QueryToOneOperator.java]: You still need this `boolean` but it should probably be called `emitDefault`. `null` is a valid default value and right now there's no way to emit it.

File: sqlbrite/src/main/java/com/squareup/sqlbrite/QueryObservable.java [hunk 2]
```diff
@@ -45,18 +45,20 @@
...
+   * emit {@code defaultValue}.
    * <p>
    * This method is equivalent to:
    * <pre>{@code
-   * flatMap(q -> q.asRows(mapper).take(1).defaultIfEmpty(null))
+   * flatMap(q -> q.asRows(mapper).take(1).defaultIfEmpty(defaultValue))
    * }</pre>
    *
    * @param mapper Maps the current {@link Cursor} row to {@code T}. May not return null.
+   * @param defaultValue value returned if result set is empty
```
Comment (JakeWharton) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/QueryObservable.java]: nit: Capitalize "Value"

File: sqlbrite/src/androidTest/java/com/squareup/sqlbrite/BriteDatabaseTest.java [hunk 3]
```diff
@@ -207,30 +207,30 @@
...
   }
 
-  @Test public void queryMapToOneOrNullEmpty() {
+  @Test public void queryMapToOneOrDefaultEmpty() {
     Employee employees = db.createQuery(TABLE_EMPLOYEE, SELECT_EMPLOYEES + " WHERE 1=2")
-        .mapToOneOrNull(Employee.MAPPER)
+        .mapToOneOrDefault(Employee.MAPPER, null)
         .toBlocking()
         .first();
     assertThat(employees).isNull();
```
Comment (JakeWharton) [inline sqlbrite/src/androidTest/java/com/squareup/sqlbrite/BriteDatabaseTest.java]: Can you duplicate this test and use a non-null default? Maybe rename this one to "EmptyUsingNull" or something.

File: sqlbrite/src/main/java/com/squareup/sqlbrite/QueryToOneOperator.java (hunk 1 above)
Comment (phajduk) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/QueryToOneOperator.java]: We can still emit `null` by passing it to the `mapToOneOrDefault` method (`null` passed as a second argument). Eventually, there could be two methods `mapToOneOrDefault` and `mapToOneOrNull` as it was previously.

Comment (JakeWharton) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/QueryToOneOperator.java]: But then you break `mapToOne` which does not emit anything on empty.

Comment (phajduk) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/QueryToOneOperator.java]: Yeah. Fixed.

File: sqlbrite/src/androidTest/java/com/squareup/sqlbrite/BriteDatabaseTest.java (hunk 3 above)
Comment (phajduk) [inline sqlbrite/src/androidTest/java/com/squareup/sqlbrite/BriteDatabaseTest.java]: Done.

File: sqlbrite/src/main/java/com/squareup/sqlbrite/QueryObservable.java (hunk 2 above)
Comment (phajduk) [inline sqlbrite/src/main/java/com/squareup/sqlbrite/QueryObservable.java]: Fixed.

**Comment by JakeWharton** [comment]
Can you squash to one commit? If not, I can do it upon merge.

**Comment by phajduk** [comment]
Squashed.

**Comment by JakeWharton** [comment]
Thanks!

**Pull request merged**